config/environments/development.rb

```diff
@@ -14,6 +14,8 @@
   # Show full error reports.
   config.consider_all_requests_local = true
 
+  config.hosts << ENV["NGROK_DOMAIN"] unless ENV["NGROK_DOMAIN"].blank?
```

Comment:
Yep, you're on the right track. `ngrok` communicates between your local machine and it's servers and exposes a proxy for your localhost at a domain that can be accessed by slack. Anyone wanting to test locally will need to recreate the app and slash command on their own slack instance to avoid  conflicting with someone else's installation.

That said, after slash commands are working I think it'd be easier most of the time to develop by allowing the HMAC to be disabled in `dev` and just use curl or whatever to hit localhost and check the response. Testing slack is good for final confirmation but the free version of ngrok is fiddly because the public domain keeps changing.